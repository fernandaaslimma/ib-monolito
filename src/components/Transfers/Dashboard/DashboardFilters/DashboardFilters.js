import React, { Fragment } from "react";
import { func } from "prop-types";

import { Filter } from "react-bocombbm-components";
import Dropdown from "../../../common/Dropdown";
import { translate } from "../../../../utils/i18n";
import { Label, FormItem } from "./styles";

import {
  PENDENCIES,
  SCHEDULED_TRANSFERS,
  ALL_TRANSFERS,
  HISTORY
} from "../../../../utils/constants";

function DashboardFilters({ onFilter, defaultFilter }) {
  const componentLabels = {
    filterButton: translate("APPLY"),
    filterLabel: translate("FILTER_BY")
  };
  return (
    <Fragment data-test="DashboardFilters">
      <Filter
        onFilter={onFilter}
        defaultFilterState={defaultFilter}
        componentLabels={componentLabels}
      >
        {(changeFunc, state) => {
          return (
            <Fragment>
              <FormItem>
                <Label>{translate("CATEGORY")}</Label>
                <Dropdown
                  onChange={changeFunc}
                  name="type"
                  list={[
                    {
                      name: translate(ALL_TRANSFERS.toUpperCase()),
                      value: ALL_TRANSFERS
                    },
                    {
                      name: translate(PENDENCIES.toUpperCase()),
                      value: PENDENCIES
                    },
                    {
                      name: translate(SCHEDULED_TRANSFERS.toUpperCase()),
                      value: SCHEDULED_TRANSFERS
                    },
                    { name: translate(HISTORY.toUpperCase()), value: HISTORY }
                  ]}
                  value={state && state.filter["type"]}
                  dataTest="DashboardCategoryFilter"
                  width={100}
                  noEmptySelection
                />
              </FormItem>
              {state && state.filter["type"] === HISTORY && (
                <FormItem>
                  <Label>{translate("PERIOD")}</Label>
                  <Dropdown
                    onChange={changeFunc}
                    name="range"
                    list={[
                      { name: translate("LAST_FIFTEEN"), value: 15 },
                      { name: translate("LAST_THIRTY"), value: 30 },
                      { name: translate("LAST_SIXTY"), value: 60 }
                    ]}
                    value={state && state.filter["range"]}
                    dataTest="DashboardRangeFilter"
                    width={100}
                    noEmptySelection
                  />
                </FormItem>
              )}
            </Fragment>
          );
        }}
      </Filter>
    </Fragment>
  );
}

DashboardFilters.propTypes = {
  onFilter: func.isRequired
};

export default DashboardFilters;
