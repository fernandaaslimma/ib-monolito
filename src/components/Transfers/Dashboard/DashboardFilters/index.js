import DashboardFilters from "./DashboardFilters";

export default DashboardFilters;
