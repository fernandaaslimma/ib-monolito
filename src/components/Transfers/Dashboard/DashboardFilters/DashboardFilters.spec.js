import React from "react";
import { render, configure } from "@testing-library/react";
import DashboardFilters from "./DashboardFilters";
configure({ testIdAttribute: "data-test" });

let props = {
  onFilter: jest.fn(),
  defaultFilter: {}
};
describe("DashboardFilters", () => {
  it("should match snapshot", () => {
    const component = render(<DashboardFilters {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should match snapshot with filter type = history", () => {
    props.defaultFilter = {
      type: "history",
      range: 20
    };
    const component = render(<DashboardFilters {...props} />);
    expect(component).toMatchSnapshot();
  });
});
