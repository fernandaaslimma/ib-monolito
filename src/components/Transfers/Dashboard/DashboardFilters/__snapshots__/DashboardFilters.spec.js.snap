// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardFilters should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htoDjs dMqmHI"
      >
        <div
          class="sc-jTzLTM eFukZp"
          data-test="FilterInfo"
        >
          <div
            class="sc-dnqmqq PzvFt"
          >
            <span
              class="sc-gzVnrw ktSBuE"
              spacing="[object Object]"
            >
              <svg
                height="16"
                viewBox="0 0 13 16"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.153 8.547a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h5.658a2.201 2.201 0 0 1 4.204 0h.485a.647.647 0 0 1 0 1.293h-.485zM9.05 6.995a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zM5.462 3.503a2.201 2.201 0 0 1-4.204 0 .647.647 0 0 1 0-1.292 2.201 2.201 0 0 1 4.204 0l6.176-.001a.647.647 0 1 1 0 1.293H5.462zM3.36 1.951a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zm3.913 11.64a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h1.778a2.201 2.201 0 0 1 4.204 0h4.365a.647.647 0 1 1 0 1.293H7.273zM5.17 12.04a.906.906 0 0 0 0 1.811.906.906 0 0 0 0-1.81z"
                  fill="currentColor"
                  fill-rule="nonzero"
                />
              </svg>
            </span>
          </div>
          <h1
            class="sc-iwsKbI kIIELH"
          >
            :
            <span
              class="sc-gZMcBi krNKDS"
            />
          </h1>
          <div
            class="sc-gqjmRU dQJWfO"
          >
            <span
              class="sc-gzVnrw ktSBuE"
              spacing="[object Object]"
            >
              <svg
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="sc-kAzzGY fWeLvu"
          data-test="FilterForm"
        >
          <div
            class="sc-fjdhpX bLppBo"
          >
            <div
              class="sc-gJWqzi bblSdI"
            >
              <label
                class="sc-jzgbtB cSIPWY"
              />
              <div
                class="sc-dymIpo fUdeBL"
                width="100"
              >
                <select
                  class="sc-cpmLhU yfqKZ"
                  data-test="DashboardCategoryFilter"
                  height="46"
                  name="type"
                  required=""
                >
                  <option
                    data-test="option_0"
                    value="all_transfers"
                  />
                  <option
                    data-test="option_1"
                    value="pendencies"
                  />
                  <option
                    data-test="option_2"
                    value="scheduled_transfers"
                  />
                  <option
                    data-test="option_3"
                    value="history"
                  />
                </select>
                <div
                  class="sc-gFaPwZ hTuBqW"
                >
                  <span
                    class="sc-fhYwyz kQFAGV"
                    height="10"
                    width="10"
                  >
                    <svg
                      fill="none"
                      height="10"
                      viewBox="0 0 14 10"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                        fill="#3976CF"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
                
              </div>
            </div>
          </div>
          <div
            class="sc-htpNat gxCRSC"
            data-test="Button"
            spacing="[object Object]"
            type="conclusive"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htoDjs dMqmHI"
    >
      <div
        class="sc-jTzLTM eFukZp"
        data-test="FilterInfo"
      >
        <div
          class="sc-dnqmqq PzvFt"
        >
          <span
            class="sc-gzVnrw ktSBuE"
            spacing="[object Object]"
          >
            <svg
              height="16"
              viewBox="0 0 13 16"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.153 8.547a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h5.658a2.201 2.201 0 0 1 4.204 0h.485a.647.647 0 0 1 0 1.293h-.485zM9.05 6.995a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zM5.462 3.503a2.201 2.201 0 0 1-4.204 0 .647.647 0 0 1 0-1.292 2.201 2.201 0 0 1 4.204 0l6.176-.001a.647.647 0 1 1 0 1.293H5.462zM3.36 1.951a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zm3.913 11.64a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h1.778a2.201 2.201 0 0 1 4.204 0h4.365a.647.647 0 1 1 0 1.293H7.273zM5.17 12.04a.906.906 0 0 0 0 1.811.906.906 0 0 0 0-1.81z"
                fill="currentColor"
                fill-rule="nonzero"
              />
            </svg>
          </span>
        </div>
        <h1
          class="sc-iwsKbI kIIELH"
        >
          :
          <span
            class="sc-gZMcBi krNKDS"
          />
        </h1>
        <div
          class="sc-gqjmRU dQJWfO"
        >
          <span
            class="sc-gzVnrw ktSBuE"
            spacing="[object Object]"
          >
            <svg
              viewBox="0 0 14 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                fill="currentColor"
                fill-rule="nonzero"
                transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-kAzzGY fWeLvu"
        data-test="FilterForm"
      >
        <div
          class="sc-fjdhpX bLppBo"
        >
          <div
            class="sc-gJWqzi bblSdI"
          >
            <label
              class="sc-jzgbtB cSIPWY"
            />
            <div
              class="sc-dymIpo fUdeBL"
              width="100"
            >
              <select
                class="sc-cpmLhU yfqKZ"
                data-test="DashboardCategoryFilter"
                height="46"
                name="type"
                required=""
              >
                <option
                  data-test="option_0"
                  value="all_transfers"
                />
                <option
                  data-test="option_1"
                  value="pendencies"
                />
                <option
                  data-test="option_2"
                  value="scheduled_transfers"
                />
                <option
                  data-test="option_3"
                  value="history"
                />
              </select>
              <div
                class="sc-gFaPwZ hTuBqW"
              >
                <span
                  class="sc-fhYwyz kQFAGV"
                  height="10"
                  width="10"
                >
                  <svg
                    fill="none"
                    height="10"
                    viewBox="0 0 14 10"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                      fill="#3976CF"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
              
            </div>
          </div>
        </div>
        <div
          class="sc-htpNat gxCRSC"
          data-test="Button"
          spacing="[object Object]"
          type="conclusive"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DashboardFilters should match snapshot with filter type = history 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htoDjs dMqmHI"
      >
        <div
          class="sc-jTzLTM eFukZp"
          data-test="FilterInfo"
        >
          <div
            class="sc-dnqmqq PzvFt"
          >
            <span
              class="sc-gzVnrw ktSBuE"
              spacing="[object Object]"
            >
              <svg
                height="16"
                viewBox="0 0 13 16"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.153 8.547a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h5.658a2.201 2.201 0 0 1 4.204 0h.485a.647.647 0 0 1 0 1.293h-.485zM9.05 6.995a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zM5.462 3.503a2.201 2.201 0 0 1-4.204 0 .647.647 0 0 1 0-1.292 2.201 2.201 0 0 1 4.204 0l6.176-.001a.647.647 0 1 1 0 1.293H5.462zM3.36 1.951a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zm3.913 11.64a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h1.778a2.201 2.201 0 0 1 4.204 0h4.365a.647.647 0 1 1 0 1.293H7.273zM5.17 12.04a.906.906 0 0 0 0 1.811.906.906 0 0 0 0-1.81z"
                  fill="currentColor"
                  fill-rule="nonzero"
                />
              </svg>
            </span>
          </div>
          <h1
            class="sc-iwsKbI kIIELH"
          >
            :
            <span
              class="sc-gZMcBi krNKDS"
            />
          </h1>
          <div
            class="sc-gqjmRU dQJWfO"
          >
            <span
              class="sc-gzVnrw ktSBuE"
              spacing="[object Object]"
            >
              <svg
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="sc-kAzzGY fWeLvu"
          data-test="FilterForm"
        >
          <div
            class="sc-fjdhpX bLppBo"
          >
            <div
              class="sc-gJWqzi bblSdI"
            >
              <label
                class="sc-jzgbtB cSIPWY"
              />
              <div
                class="sc-dymIpo fUdeBL"
                width="100"
              >
                <select
                  class="sc-cpmLhU yfqKZ"
                  data-test="DashboardCategoryFilter"
                  height="46"
                  name="type"
                  required=""
                >
                  <option
                    data-test="option_0"
                    value="all_transfers"
                  />
                  <option
                    data-test="option_1"
                    value="pendencies"
                  />
                  <option
                    data-test="option_2"
                    value="scheduled_transfers"
                  />
                  <option
                    data-test="option_3"
                    value="history"
                  />
                </select>
                <div
                  class="sc-gFaPwZ hTuBqW"
                >
                  <span
                    class="sc-fhYwyz kQFAGV"
                    height="10"
                    width="10"
                  >
                    <svg
                      fill="none"
                      height="10"
                      viewBox="0 0 14 10"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                        fill="#3976CF"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
                
              </div>
            </div>
            <div
              class="sc-gJWqzi bblSdI"
            >
              <label
                class="sc-jzgbtB cSIPWY"
              />
              <div
                class="sc-dymIpo fUdeBL"
                width="100"
              >
                <select
                  class="sc-cpmLhU yfqKZ"
                  data-test="DashboardRangeFilter"
                  height="46"
                  name="range"
                  required=""
                >
                  <option
                    data-test="option_0"
                    value="15"
                  />
                  <option
                    data-test="option_1"
                    value="30"
                  />
                  <option
                    data-test="option_2"
                    value="60"
                  />
                </select>
                <div
                  class="sc-gFaPwZ hTuBqW"
                >
                  <span
                    class="sc-fhYwyz kQFAGV"
                    height="10"
                    width="10"
                  >
                    <svg
                      fill="none"
                      height="10"
                      viewBox="0 0 14 10"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                        fill="#3976CF"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
                
              </div>
            </div>
          </div>
          <div
            class="sc-htpNat gxCRSC"
            data-test="Button"
            spacing="[object Object]"
            type="conclusive"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htoDjs dMqmHI"
    >
      <div
        class="sc-jTzLTM eFukZp"
        data-test="FilterInfo"
      >
        <div
          class="sc-dnqmqq PzvFt"
        >
          <span
            class="sc-gzVnrw ktSBuE"
            spacing="[object Object]"
          >
            <svg
              height="16"
              viewBox="0 0 13 16"
              width="13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.153 8.547a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h5.658a2.201 2.201 0 0 1 4.204 0h.485a.647.647 0 0 1 0 1.293h-.485zM9.05 6.995a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zM5.462 3.503a2.201 2.201 0 0 1-4.204 0 .647.647 0 0 1 0-1.292 2.201 2.201 0 0 1 4.204 0l6.176-.001a.647.647 0 1 1 0 1.293H5.462zM3.36 1.951a.906.906 0 0 0 0 1.81.906.906 0 0 0 0-1.81zm3.913 11.64a2.201 2.201 0 0 1-4.204 0H1.291a.647.647 0 1 1 0-1.293h1.778a2.201 2.201 0 0 1 4.204 0h4.365a.647.647 0 1 1 0 1.293H7.273zM5.17 12.04a.906.906 0 0 0 0 1.811.906.906 0 0 0 0-1.81z"
                fill="currentColor"
                fill-rule="nonzero"
              />
            </svg>
          </span>
        </div>
        <h1
          class="sc-iwsKbI kIIELH"
        >
          :
          <span
            class="sc-gZMcBi krNKDS"
          />
        </h1>
        <div
          class="sc-gqjmRU dQJWfO"
        >
          <span
            class="sc-gzVnrw ktSBuE"
            spacing="[object Object]"
          >
            <svg
              viewBox="0 0 14 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                fill="currentColor"
                fill-rule="nonzero"
                transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="sc-kAzzGY fWeLvu"
        data-test="FilterForm"
      >
        <div
          class="sc-fjdhpX bLppBo"
        >
          <div
            class="sc-gJWqzi bblSdI"
          >
            <label
              class="sc-jzgbtB cSIPWY"
            />
            <div
              class="sc-dymIpo fUdeBL"
              width="100"
            >
              <select
                class="sc-cpmLhU yfqKZ"
                data-test="DashboardCategoryFilter"
                height="46"
                name="type"
                required=""
              >
                <option
                  data-test="option_0"
                  value="all_transfers"
                />
                <option
                  data-test="option_1"
                  value="pendencies"
                />
                <option
                  data-test="option_2"
                  value="scheduled_transfers"
                />
                <option
                  data-test="option_3"
                  value="history"
                />
              </select>
              <div
                class="sc-gFaPwZ hTuBqW"
              >
                <span
                  class="sc-fhYwyz kQFAGV"
                  height="10"
                  width="10"
                >
                  <svg
                    fill="none"
                    height="10"
                    viewBox="0 0 14 10"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                      fill="#3976CF"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
              
            </div>
          </div>
          <div
            class="sc-gJWqzi bblSdI"
          >
            <label
              class="sc-jzgbtB cSIPWY"
            />
            <div
              class="sc-dymIpo fUdeBL"
              width="100"
            >
              <select
                class="sc-cpmLhU yfqKZ"
                data-test="DashboardRangeFilter"
                height="46"
                name="range"
                required=""
              >
                <option
                  data-test="option_0"
                  value="15"
                />
                <option
                  data-test="option_1"
                  value="30"
                />
                <option
                  data-test="option_2"
                  value="60"
                />
              </select>
              <div
                class="sc-gFaPwZ hTuBqW"
              >
                <span
                  class="sc-fhYwyz kQFAGV"
                  height="10"
                  width="10"
                >
                  <svg
                    fill="none"
                    height="10"
                    viewBox="0 0 14 10"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12.2229 1.23459L7.00847 6.64326L1.79283 1.23335C1.60427 1.03776 1.3561 0.940155 1.10556 0.940155C0.856838 0.940155 0.605267 1.03651 0.408109 1.23079C0.219243 1.42662 0.125001 1.68575 0.125001 1.94542C0.125001 2.20509 0.219243 2.46422 0.405719 2.65764L6.31484 8.76931C6.50341 8.9649 6.75157 9.0625 7.00212 9.0625C7.25142 9.0625 7.5036 8.96569 7.70096 8.77048L13.5944 2.63993C13.7808 2.44666 13.875 2.18753 13.875 1.92786C13.875 1.66819 13.7808 1.40906 13.5944 1.21578C13.4043 1.01854 13.1535 0.92109 12.9035 0.924849C12.6571 0.928553 12.4106 1.03041 12.2229 1.23459Z"
                      fill="#3976CF"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
              
            </div>
          </div>
        </div>
        <div
          class="sc-htpNat gxCRSC"
          data-test="Button"
          spacing="[object Object]"
          type="conclusive"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
